### PR TITLE
dev(codespaces): add basic code spaces setup

### PR DESCRIPTION
**Pull request by hazzadous**
This just uses the github codespaces "universal" image, which has things
like kubectl, helm etc. and then adds in kind and creates a cluster.

I'm using the universal image with the hope that this will be caches on
the codespaces nodes :fingerscrossed:

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## How has this been tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by guidoiaquinti — APPROVED** [review]
Non-blocking: can we add some info in the `REDAME.md` on how to use it?

File: .devcontainer/setup_container.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    protocol: TCP
+  - containerPort: 443
+    hostPort: 4430
+    protocol: TCP
+EOF
+
+# Then provision the nginx controller, e.g.
+# https://kind.sigs.k8s.io/docs/user/ingress/#ingress-nginx 
+# NOTE: I've pinned the version here
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/d8c9a6c238f714587da4d2ac2dcd0d3d39419ccf/deploy/static/provider/kind/deploy.yaml
```
Comment (guidoiaquinti) [inline .devcontainer/setup_container.sh:43]: Do we always need this?

File: .devcontainer/setup_container.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+echo "printf 'Hello 🦔! To install PostHog into k8s run this:\n\n "helm upgrade --install posthog charts/posthog --set cloud=private"\n'" >> ~/.zshrc
```
Comment (guidoiaquinti) [inline .devcontainer/setup_container.sh:45]: This will not work outside `zsh`

File: .devcontainer/setup_container.sh (hunk 1 above)
Comment (hazzadous) [reply .devcontainer/setup_container.sh:43]: possibly not 🤔

File: .devcontainer/setup_container.sh (hunk 2 above)
Comment (hazzadous) [reply .devcontainer/setup_container.sh:45]: The image user is setup for zsh, I can change if you think useful?

**Comment by hazzadous** [comment]
> Non-blocking: can we add some info in the `REDAME.md` on how to use it?

Sure!

File: .devcontainer/setup_container.sh (hunk 2 above)
Comment (guidoiaquinti) [reply .devcontainer/setup_container.sh:45]: Sorry, in that case make sense

**Pull request merged**